src/screens/DirectMessagesScreen.tsx

```diff
@@ -10,26 +10,36 @@ import { createStyles } from '../components';
 import { ActivityIndicator } from 'react-native-paper';
 import { HomeStackScreenProps } from '../navigators/types';
 import { t } from 'i18next';
+import { useAsyncStorage } from '../hooks/useAsyncStorage';
 
 export function DirectMessagesScreen({
   navigation,
   route,
 }: HomeStackScreenProps<'Home/DirectMessage'>) {
   const { recipientUserId } = route.params;
+  const { styles } = useStyles(defaultStyles);
 
   useLayoutEffect(() => {
     navigation.setOptions({
       title: t('direct-messages-title', { userId: recipientUserId }),
     });
   }, [navigation, recipientUserId]);
-  const { styles } = useStyles(defaultStyles);
 
   const { data: userData } = useUser();
   const { data, isLoading, hasNextPage, fetchNextPage, isFetchingNextPage } =
     useInfinitePrivatePosts(recipientUserId);
-  const incomingMessages = data?.pages?.flatMap((page) =>
-    page.privatePosts.edges.flatMap((edge) => postToMessage(edge.node)),
+  const [_, setEndCursor] = useAsyncStorage(
+    `${userData?.id}-${recipientUserId}-endCursor`,
   );
```

Comment:
Instead of using asyncStorage directly here, can we encapsulate how cursors are stored (and this formatted key) into a helper method, perhaps exposed from `useDirectMessageEndCursor` hook/file?

Reply:
Done, I already a helper for this but failed to utilize it here.